backends/rabbitmq-streams/write.go

```diff
@@ -0,0 +1,65 @@
+package rabbitmqStreams
+
+import (
+	"github.com/rabbitmq/rabbitmq-stream-go-client/pkg/stream"
+
+	"github.com/jhump/protoreflect/desc"
+	"github.com/pkg/errors"
+	"github.com/rabbitmq/rabbitmq-stream-go-client/pkg/amqp"
+	"github.com/sirupsen/logrus"
+
+	"github.com/batchcorp/plumber/cli"
+	"github.com/batchcorp/plumber/writer"
+)
+
+func Write(opts *cli.Options, md *desc.MessageDescriptor) error {
+
+	writeValues, err := writer.GenerateWriteValues(md, opts)
+	if err != nil {
+		return errors.Wrap(err, "unable to generate write value")
+	}
+
+	client, err := NewClient(opts)
+	if err != nil {
+		return err
+	}
+
+	defer client.Close()
+
+	producer, err := client.NewProducer(opts.RabbitMQStreams.Stream, &stream.ProducerOptions{
+		Name:      opts.RabbitMQStreams.ClientName,
+		QueueSize: 0,
+		BatchSize: len(writeValues),
+	})
+	if err != nil {
+		return errors.Wrap(err, "unable to create rabbitmq streams producer")
+	}
+
+	defer producer.Close()
+
+	r := &RabbitMQStreams{
+		Client:   client,
+		Producer: producer,
+		Options:  opts,
+		log:      logrus.WithField("pkg", "rabbitmq-streams/write.go"),
+	}
+
+	for _, value := range writeValues {
+		if err := r.Write(value); err != nil {
+			r.log.Error(err)
+			continue
+		}
+	}
+
+	return nil
+}
+
+func (r *RabbitMQStreams) Write(value []byte) error {
+	if err := r.Producer.Send(amqp.NewMessage(value)); err != nil {
```

Comment:
I would add also the handlers to publish confirmation like:
https://github.com/rabbitmq/rabbitmq-stream-go-client/blob/main/examples/getting_started.go#L75-L77
and publish errors:
https://github.com/rabbitmq/rabbitmq-stream-go-client/blob/main/examples/publishersError/publisherError.go#L57

I think that at some point I will aggregate them in one single handler

Reply:
Just to let you know:
There is also a work in progress [ha producer](https://github.com/rabbitmq/rabbitmq-stream-go-client/blob/main/examples/haProducer/producer.go)

Reply:
Good catch! Added these